New UI for HomePage

xinyao-song opened the pull request:
Added a new scroll and UI for home
and basically fixed a bunch of issues and features with follow all and stuff 
check ALL features to see if it works

REMEMBER run all of these:
npm cache clean --force
rm -rf node_modules package-lock.json
npm install
npx react-native-asset

xinyao-song committed:
xs - resolved merge conflict to main

AllenHsm reviewed (approved):
Perfect! The search bar disappear when scrolling down!

AllenHsm commented on GOLDTracker/src/components/SearchComponent.js:
```diff
@@ -1,24 +1,93 @@
-import CategorySearch from "../components/CategorySearch";
 import * as ClassRegister from "./ClassRegister";
+import { COLORS } from "../theme/theme";
+import departmentMapping from "../assets/departmentMapping.json";
```
should replace departmentMapping with departmentList.json

AllenHsm commented on GOLDTracker/src/components/SearchComponent.js:
```diff
@@ -42,95 +111,172 @@ const SearchComponent = ({ search, setSearch, setIsSearching, major }) => {
-      if (deptCode){
-        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)
-
+      const apiUrl = /^[A-Z]{2,}\s[\dA-Z]+$/.test(searchTerm)
```
For subject area search and department search, we should use two different parameters in API. You can see the difference in current main branch. 

      if (deptCode){
        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)

        ? `${API_URL}?quarter=${quarter}&courseId=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true`
        : `${API_URL}?quarter=${quarter}&deptCode=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true&pageNumber=1&pageSize=30`;
      }
      else{
        apiUrl = /^[A-Z|a-z]{2,}\s[\d(A-Z|a-z]+$/.test(searchTerm)
        ? `${API_URL}?quarter=${quarter}&courseId=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true`
        : `${API_URL}?quarter=${quarter}&subjectCode=${encodeURIComponent(
            searchTerm
          )}&includeClassSections=true&pageNumber=1&pageSize=30`;
      }

AllenHsm commented on GOLDTracker/src/components/SearchComponent.js:
```diff
@@ -1,24 +1,93 @@
+    { code: "Your Major", label: "Cancel Selection" },
+    ...Object.keys(departmentMapping).map((code) => ({
+      code,
+      label: departmentMapping[code][0],
```
remove the [0] here to fit the new mapping json. Should be: label: departmentMapping[code]

AllenHsm reviewed (changes requested):
Need to correct several errors causing it unable to search subject areas like "CHIN" and the wrong department list.

AllenHsm committed:
Update SearchComponent.js

Change the search by department and api setting back to the correct version.

AllenHsm committed:
Update SearchComponent.js

Update the placeholder, since now we have an individual search by department component.

AllenHsm committed:
Merge branch 'main' of github.com:ucsb-cs184-f24/team04-Goldxxx into newFency

AllenHsm committed:
Merge branch 'main' of github.com:ucsb-cs184-f24/team04-Goldxxx into newFency

xinyao-song committed:
xs - minor change for search

xinyao-song committed:
Merge branch 'newFency' of https://github.com/ucsb-cs184-f24/team04-GOLDTracker into newFency

xinyao-song committed:
xs - adjusted final height

xinyao-song committed:
xs - synced ICON to all be entypo

Division7 committed:
missing class.js changes

Division7 committed:
fixed follow buttons, cart screen

xinyao-song committed:
xs - Modified UI for course detail and developer options

xinyao-song committed:
xs - resolve merge conflict

AllenHsm reviewed (approved):
LGTM!!!

Simonmann17 reviewed (approved):
LGTM!
